api/tests/opentrons/drivers/test_utils.py

```
@@ -0,0 +1,104 @@
+from typing import Dict
+import pytest
+from opentrons.drivers import utils
+from opentrons.drivers.types import Temperature, PlateTemperature
+
+
+@pytest.mark.parametrize(
+    argnames=['input_str', 'expected_result'],
+    argvalues=[
+        ['version:123-2 serial:serial_v model:m',
+         {'version': '123-2', 'serial': 'serial_v', 'model': 'm'}],
+        ['serial:serial_v model:m version:123-2 ',
+         {'version': '123-2', 'serial': 'serial_v', 'model': 'm'}],
+        ['   serial:serial_v model:m    version:123-2   ',
+         {'version': '123-2', 'serial': 'serial_v', 'model': 'm'}]
+    ]
+)
+def test_parse_device_information_success(
+        input_str: str, expected_result: Dict[str, str]) -> None:
+    """Test parse device information."""
+    assert utils.parse_device_information(input_str) == expected_result
```

Should we also test the current (I think) behavior that extra `k:v` pairs are silently ignored?

```python
['serial:serial_v extra_k:extra_v model:m version:123-2',
 {'version': '123-2', 'serial': 'serial_v', 'model': 'm'}]
```

I assume this is intended, but I don't know how important it is.

---

Sure. Why not.

---

Thanks. I realize that the caller was responsible for ignoring extra keys. I changed it to omit extra keys.